### ISSUE TITLE
Allow all pick field options to be filtered

### ISSUE THREAD

**Issue by JoryHogeveen** [pods-framework/pods#5353]
**Is your feature request related to a problem? Please describe.**
Currently it's only possible the change the available field on the UI side, not the backend side.
This results in weird behavior when changing options in pick fields like saving doesn't work.

Currently the only filter available are these:

- `pods_field_pick_object_data`
- `pods_form_ui_field_pick_custom_values` (custom-simple type only)

**Describe the solution you'd like**
Add a filter before Pick field (any type) returns the available options.

**Additional context**

**Case 1:**
Simple relationship (custom defined list).

I'd like to define the values available through code instead of field settings.
In this case it's a select box in which the options should be the available theme hooks (like Genesis). Though when using another theme these hooks will be different.
So: I just create a pick field without any predefined options and would like to enable these options through a filter.

**Case 2:**
Another (previous) case is that a pick field relationship values is showing all related object and I wanted to filter the available options so I can remove some of them based on the current user.

Specific example:
Filter the available Post relationships based on the author.
(I know this is a simple example and it could be done with the WHERE option)
